// @mantine/core/esm/index.js
'use client';
export { RemoveScroll } from 'react-remove-scroll';
export { keys } from './core/utils/keys/keys.js';
export { deepMerge } from './core/utils/deep-merge/deep-merge.js';
export { camelToKebabCase } from './core/utils/camel-to-kebab-case/camel-to-kebab-case.js';
export { px } from './core/utils/units-converters/px.js';
export { em, rem } from './core/utils/units-converters/rem.js';
export { filterProps } from './core/utils/filter-props/filter-props.js';
export { isNumberLike } from './core/utils/is-number-like/is-number-like.js';
export { isElement } from './core/utils/is-element/is-element.js';
export { createSafeContext } from './core/utils/create-safe-context/create-safe-context.js';
export { createOptionalContext } from './core/utils/create-optional-context/create-optional-context.js';
export { getSafeId } from './core/utils/get-safe-id/get-safe-id.js';
export { createScopedKeydownHandler } from './core/utils/create-scoped-keydown-handler/create-scoped-keydown-handler.js';
export { findElementAncestor } from './core/utils/find-element-ancestor/find-element-ancestor.js';
export { getDefaultZIndex } from './core/utils/get-default-z-index/get-default-z-index.js';
export { closeOnEscape } from './core/utils/close-on-escape/close-on-escape.js';
export { noop } from './core/utils/noop/noop.js';
export { getFontSize, getLineHeight, getRadius, getShadow, getSize, getSpacing } from './core/utils/get-size/get-size.js';
export { createEventHandler } from './core/utils/create-event-handler/create-event-handler.js';
export { getBreakpointValue } from './core/utils/get-breakpoint-value/get-breakpoint-value.js';
export { getSortedBreakpoints } from './core/utils/get-sorted-breakpoints/get-sorted-breakpoints.js';
export { getBaseValue } from './core/utils/get-base-value/get-base-value.js';
export { getContextItemIndex } from './core/utils/get-context-item-index/get-context-item-index.js';
export { useHovered } from './core/utils/use-hovered/use-hovered.js';
export { createUseExternalEvents } from './core/utils/create-use-external-events/create-use-external-events.js';
export { createVarsResolver } from './core/styles-api/create-vars-resolver/create-vars-resolver.js';
export { resolveClassNames } from './core/styles-api/use-styles/get-class-name/resolve-class-names/resolve-class-names.js';
export { resolveStyles } from './core/styles-api/use-styles/get-style/resolve-styles/resolve-styles.js';
export { useResolvedStylesApi } from './core/styles-api/use-resolved-styles-api/use-resolved-styles-api.js';
export { FOCUS_CLASS_NAMES } from './core/styles-api/use-styles/get-class-name/get-global-class-names/get-global-class-names.js';
export { useStyles } from './core/styles-api/use-styles/use-styles.js';
export { DEFAULT_THEME } from './core/MantineProvider/default-theme.js';
export { mergeMantineTheme, validateMantineTheme } from './core/MantineProvider/merge-mantine-theme/merge-mantine-theme.js';
export { MantineProvider } from './core/MantineProvider/MantineProvider.js';
export { MantineContext, useMantineClassNamesPrefix, useMantineContext, useMantineCssVariablesResolver, useMantineStyleNonce } from './core/MantineProvider/Mantine.context.js';
export { MantineThemeContext, MantineThemeProvider, useMantineTheme, useSafeMantineTheme } from './core/MantineProvider/MantineThemeProvider/MantineThemeProvider.js';
export { useProps } from './core/MantineProvider/use-props/use-props.js';
export { convertCssVariables } from './core/MantineProvider/convert-css-variables/convert-css-variables.js';
export { createTheme } from './core/MantineProvider/create-theme/create-theme.js';
export { mergeThemeOverrides } from './core/MantineProvider/merge-theme-overrides/merge-theme-overrides.js';
export { defaultCssVariablesResolver } from './core/MantineProvider/MantineCssVariables/default-css-variables-resolver.js';
export { localStorageColorSchemeManager } from './core/MantineProvider/color-scheme-managers/local-storage-manager.js';
export { isMantineColorScheme } from './core/MantineProvider/color-scheme-managers/is-mantine-color-scheme.js';
export { getPrimaryShade } from './core/MantineProvider/color-functions/get-primary-shade/get-primary-shade.js';
export { parseThemeColor } from './core/MantineProvider/color-functions/parse-theme-color/parse-theme-color.js';
export { getThemeColor } from './core/MantineProvider/color-functions/get-theme-color/get-theme-color.js';
export { defaultVariantColorsResolver } from './core/MantineProvider/color-functions/default-variant-colors-resolver/default-variant-colors-resolver.js';
export { getGradient } from './core/MantineProvider/color-functions/get-gradient/get-gradient.js';
export { toRgba } from './core/MantineProvider/color-functions/to-rgba/to-rgba.js';
export { rgba } from './core/MantineProvider/color-functions/rgba/rgba.js';
export { darken } from './core/MantineProvider/color-functions/darken/darken.js';
export { lighten } from './core/MantineProvider/color-functions/lighten/lighten.js';
export { isLightColor } from './core/MantineProvider/color-functions/is-light-color/is-light-color.js';
export { useMantineColorScheme } from './core/MantineProvider/use-mantine-color-scheme/use-mantine-color-scheme.js';
export { useProviderColorScheme } from './core/MantineProvider/use-mantine-color-scheme/use-provider-color-scheme.js';
export { useComputedColorScheme } from './core/MantineProvider/use-mantine-color-scheme/use-computed-color-scheme.js';
export { ColorSchemeScript } from './core/MantineProvider/ColorSchemeScript/ColorSchemeScript.js';
export { InlineStyles } from './core/InlineStyles/InlineStyles.js';
export { stylesToString } from './core/InlineStyles/styles-to-string/styles-to-string.js';
export { useRandomClassName } from './core/Box/use-random-classname/use-random-classname.js';
export { getStyleObject } from './core/Box/get-style-object/get-style-object.js';
export { Box } from './core/Box/Box.js';
export { extractStyleProps } from './core/Box/style-props/extract-style-props/extract-style-props.js';
export { STYlE_PROPS_DATA } from './core/Box/style-props/style-props-data.js';
export { parseStyleProps } from './core/Box/style-props/parse-style-props/parse-style-props.js';
export { factory } from './core/factory/factory.js';
export { polymorphicFactory } from './core/factory/polymorphic-factory.js';
export { createPolymorphicComponent } from './core/factory/create-polymorphic-component.js';
export { DirectionContext, DirectionProvider, useDirection } from './core/DirectionProvider/DirectionProvider.js';
export { Collapse } from './components/Collapse/Collapse.js';
export { ScrollArea, ScrollAreaAutosize } from './components/ScrollArea/ScrollArea.js';
export { UnstyledButton } from './components/UnstyledButton/UnstyledButton.js';
export { VisuallyHidden } from './components/VisuallyHidden/VisuallyHidden.js';
export { Paper } from './components/Paper/Paper.js';
export { Popover } from './components/Popover/Popover.js';
export { PopoverDropdown } from './components/Popover/PopoverDropdown/PopoverDropdown.js';
export { PopoverTarget } from './components/Popover/PopoverTarget/PopoverTarget.js';
export { ActionIcon } from './components/ActionIcon/ActionIcon.js';
export { ActionIconGroup } from './components/ActionIcon/ActionIconGroup/ActionIconGroup.js';
export { CloseIcon } from './components/CloseButton/CloseIcon.js';
export { CloseButton } from './components/CloseButton/CloseButton.js';
export { Group } from './components/Group/Group.js';
export { Loader, defaultLoaders } from './components/Loader/Loader.js';
export { Overlay } from './components/Overlay/Overlay.js';
export { ModalBase } from './components/ModalBase/ModalBase.js';
export { ModalBaseBody } from './components/ModalBase/ModalBaseBody.js';
export { ModalBaseCloseButton } from './components/ModalBase/ModalBaseCloseButton.js';
export { ModalBaseContent } from './components/ModalBase/ModalBaseContent.js';
export { ModalBaseHeader } from './components/ModalBase/ModalBaseHeader.js';
export { ModalBaseOverlay } from './components/ModalBase/ModalBaseOverlay.js';
export { ModalBaseTitle } from './components/ModalBase/ModalBaseTitle.js';
export { NativeScrollArea } from './components/ModalBase/NativeScrollArea.js';
export { Input } from './components/Input/Input.js';
export { InputWrapper } from './components/Input/InputWrapper/InputWrapper.js';
export { InputDescription } from './components/Input/InputDescription/InputDescription.js';
export { InputError } from './components/Input/InputError/InputError.js';
export { InputLabel } from './components/Input/InputLabel/InputLabel.js';
export { InputPlaceholder } from './components/Input/InputPlaceholder/InputPlaceholder.js';
export { useInputProps } from './components/Input/use-input-props.js';
export { useInputWrapperContext } from './components/Input/InputWrapper.context.js';
export { InputBase } from './components/InputBase/InputBase.js';
export { FLEX_STYLE_PROPS_DATA } from './components/Flex/flex-props.js';
export { Flex } from './components/Flex/Flex.js';
export { Accordion } from './components/Accordion/Accordion.js';
export { AccordionChevron } from './components/Accordion/AccordionChevron.js';
export { AccordionItem } from './components/Accordion/AccordionItem/AccordionItem.js';
export { AccordionPanel } from './components/Accordion/AccordionPanel/AccordionPanel.js';
export { AccordionControl } from './components/Accordion/AccordionControl/AccordionControl.js';
export { Affix } from './components/Affix/Affix.js';
export { Alert } from './components/Alert/Alert.js';
export { Anchor } from './components/Anchor/Anchor.js';
export { AppShell } from './components/AppShell/AppShell.js';
export { AppShellAside } from './components/AppShell/AppShellAside/AppShellAside.js';
export { AppShellFooter } from './components/AppShell/AppShellFooter/AppShellFooter.js';
export { AppShellHeader } from './components/AppShell/AppShellHeader/AppShellHeader.js';
export { AppShellNavbar } from './components/AppShell/AppShellNavbar/AppShellNavbar.js';
export { AppShellSection } from './components/AppShell/AppShellSection/AppShellSection.js';
export { AppShellMain } from './components/AppShell/AppShellMain/AppShellMain.js';
export { AspectRatio } from './components/AspectRatio/AspectRatio.js';
export { Autocomplete } from './components/Autocomplete/Autocomplete.js';
export { Avatar } from './components/Avatar/Avatar.js';
export { AvatarGroup } from './components/Avatar/AvatarGroup/AvatarGroup.js';
export { BackgroundImage } from './components/BackgroundImage/BackgroundImage.js';
export { Badge } from './components/Badge/Badge.js';
export { Blockquote } from './components/Blockquote/Blockquote.js';
export { Breadcrumbs } from './components/Breadcrumbs/Breadcrumbs.js';
export { Burger } from './components/Burger/Burger.js';
export { Button } from './components/Button/Button.js';
export { ButtonGroup } from './components/Button/ButtonGroup/ButtonGroup.js';
export { Card } from './components/Card/Card.js';
export { CardSection } from './components/Card/CardSection/CardSection.js';
export { Center } from './components/Center/Center.js';
export { Checkbox } from './components/Checkbox/Checkbox.js';
export { CheckboxGroup } from './components/Checkbox/CheckboxGroup/CheckboxGroup.js';
export { CheckIcon } from './components/Checkbox/CheckIcon.js';
export { Chip } from './components/Chip/Chip.js';
export { ChipGroup } from './components/Chip/ChipGroup/ChipGroup.js';
export { Code } from './components/Code/Code.js';
export { ColorPicker } from './components/ColorPicker/ColorPicker.js';
export { convertHsvaTo } from './components/ColorPicker/converters/converters.js';
export { isColorValid, parseColor } from './components/ColorPicker/converters/parsers.js';
export { ColorInput } from './components/ColorInput/ColorInput.js';
export { ColorSwatch } from './components/ColorSwatch/ColorSwatch.js';
export { getParsedComboboxData } from './components/Combobox/get-parsed-combobox-data/get-parsed-combobox-data.js';
export { getLabelsLockup, getOptionsLockup } from './components/Combobox/get-options-lockup/get-options-lockup.js';
export { ComboboxChevron } from './components/Combobox/ComboboxChevron/ComboboxChevron.js';
export { Combobox } from './components/Combobox/Combobox.js';
export { ComboboxDropdown } from './components/Combobox/ComboboxDropdown/ComboboxDropdown.js';
export { ComboboxOptions } from './components/Combobox/ComboboxOptions/ComboboxOptions.js';
export { ComboboxOption } from './components/Combobox/ComboboxOption/ComboboxOption.js';
export { ComboboxTarget } from './components/Combobox/ComboboxTarget/ComboboxTarget.js';
export { ComboboxSearch } from './components/Combobox/ComboboxSearch/ComboboxSearch.js';
export { ComboboxEmpty } from './components/Combobox/ComboboxEmpty/ComboboxEmpty.js';
export { ComboboxFooter } from './components/Combobox/ComboboxFooter/ComboboxFooter.js';
export { ComboboxHeader } from './components/Combobox/ComboboxHeader/ComboboxHeader.js';
export { ComboboxEventsTarget } from './components/Combobox/ComboboxEventsTarget/ComboboxEventsTarget.js';
export { ComboboxDropdownTarget } from './components/Combobox/ComboboxDropdownTarget/ComboboxDropdownTarget.js';
export { ComboboxGroup } from './components/Combobox/ComboboxGroup/ComboboxGroup.js';
export { OptionsDropdown } from './components/Combobox/OptionsDropdown/OptionsDropdown.js';
export { useCombobox } from './components/Combobox/use-combobox/use-combobox.js';
export { useVirtualizedCombobox } from './components/Combobox/use-combobox/use-virtualized-combobox.js';
export { useComboboxTargetProps } from './components/Combobox/use-combobox-target-props/use-combobox-target-props.js';
export { isOptionsGroup } from './components/Combobox/OptionsDropdown/is-options-group.js';
export { defaultOptionsFilter } from './components/Combobox/OptionsDropdown/default-options-filter.js';
export { Container } from './components/Container/Container.js';
export { CopyButton } from './components/CopyButton/CopyButton.js';
export { Dialog } from './components/Dialog/Dialog.js';
export { Divider } from './components/Divider/Divider.js';
export { Drawer } from './components/Drawer/Drawer.js';
export { DrawerRoot } from './components/Drawer/DrawerRoot.js';
export { DrawerBody } from './components/Drawer/DrawerBody.js';
export { DrawerCloseButton } from './components/Drawer/DrawerCloseButton.js';
export { DrawerContent } from './components/Drawer/DrawerContent.js';
export { DrawerHeader } from './components/Drawer/DrawerHeader.js';
export { DrawerOverlay } from './components/Drawer/DrawerOverlay.js';
export { DrawerTitle } from './components/Drawer/DrawerTitle.js';
export { Fieldset } from './components/Fieldset/Fieldset.js';
export { FileButton } from './components/FileButton/FileButton.js';
export { FileInput } from './components/FileInput/FileInput.js';
export { useDelayedHover } from './components/Floating/use-delayed-hover.js';
export { useFloatingAutoUpdate } from './components/Floating/use-floating-auto-update.js';
export { getFloatingPosition } from './components/Floating/get-floating-position/get-floating-position.js';
export { FloatingArrow } from './components/Floating/FloatingArrow/FloatingArrow.js';
export { FocusTrap } from './components/FocusTrap/FocusTrap.js';
export { Grid } from './components/Grid/Grid.js';
export { GridCol } from './components/Grid/GridCol/GridCol.js';
export { Highlight } from './components/Highlight/Highlight.js';
export { HoverCard } from './components/HoverCard/HoverCard.js';
export { HoverCardDropdown } from './components/HoverCard/HoverCardDropdown/HoverCardDropdown.js';
export { HoverCardTarget } from './components/HoverCard/HoverCardTarget/HoverCardTarget.js';
export { Image } from './components/Image/Image.js';
export { Indicator } from './components/Indicator/Indicator.js';
export { JsonInput } from './components/JsonInput/JsonInput.js';
export { Kbd } from './components/Kbd/Kbd.js';
export { List } from './components/List/List.js';
export { ListItem } from './components/List/ListItem/ListItem.js';
export { LoadingOverlay } from './components/LoadingOverlay/LoadingOverlay.js';
export { Mark } from './components/Mark/Mark.js';
export { Menu } from './components/Menu/Menu.js';
export { MenuDivider } from './components/Menu/MenuDivider/MenuDivider.js';
export { MenuItem } from './components/Menu/MenuItem/MenuItem.js';
export { MenuLabel } from './components/Menu/MenuLabel/MenuLabel.js';
export { MenuDropdown } from './components/Menu/MenuDropdown/MenuDropdown.js';
export { MenuTarget } from './components/Menu/MenuTarget/MenuTarget.js';
export { Modal } from './components/Modal/Modal.js';
export { ModalRoot } from './components/Modal/ModalRoot.js';
export { ModalBody } from './components/Modal/ModalBody.js';
export { ModalCloseButton } from './components/Modal/ModalCloseButton.js';
export { ModalContent } from './components/Modal/ModalContent.js';
export { ModalHeader } from './components/Modal/ModalHeader.js';
export { ModalOverlay } from './components/Modal/ModalOverlay.js';
export { ModalTitle } from './components/Modal/ModalTitle.js';
export { MultiSelect } from './components/MultiSelect/MultiSelect.js';
export { NativeSelect } from './components/NativeSelect/NativeSelect.js';
export { NavLink } from './components/NavLink/NavLink.js';
export { Notification } from './components/Notification/Notification.js';
export { NumberInput } from './components/NumberInput/NumberInput.js';
export { Pagination } from './components/Pagination/Pagination.js';
export { PaginationControl } from './components/Pagination/PaginationControl/PaginationControl.js';
export { PaginationDots } from './components/Pagination/PaginationDots/PaginationDots.js';
export { PaginationFirst, PaginationLast, PaginationNext, PaginationPrevious } from './components/Pagination/PaginationEdges/PaginationEdges.js';
export { PaginationItems } from './components/Pagination/PaginationItems/PaginationItems.js';
export { PaginationRoot } from './components/Pagination/PaginationRoot/PaginationRoot.js';
export { PasswordInput } from './components/PasswordInput/PasswordInput.js';
export { Pill } from './components/Pill/Pill.js';
export { PillGroup } from './components/Pill/PillGroup/PillGroup.js';
export { PillsInput } from './components/PillsInput/PillsInput.js';
export { PillsInputField } from './components/PillsInput/PillsInputField/PillsInputField.js';
export { PinInput } from './components/PinInput/PinInput.js';
export { Portal } from './components/Portal/Portal.js';
export { OptionalPortal } from './components/Portal/OptionalPortal.js';
export { Progress } from './components/Progress/Progress.js';
export { ProgressLabel } from './components/Progress/ProgressLabel/ProgressLabel.js';
export { ProgressRoot } from './components/Progress/ProgressRoot/ProgressRoot.js';
export { ProgressSection } from './components/Progress/ProgressSection/ProgressSection.js';
export { Radio } from './components/Radio/Radio.js';
export { RadioGroup } from './components/Radio/RadioGroup/RadioGroup.js';
export { RadioIcon } from './components/Radio/RadioIcon.js';
export { Rating } from './components/Rating/Rating.js';
export { RingProgress } from './components/RingProgress/RingProgress.js';
export { SegmentedControl } from './components/SegmentedControl/SegmentedControl.js';
export { Select } from './components/Select/Select.js';
export { SimpleGrid } from './components/SimpleGrid/SimpleGrid.js';
export { Skeleton } from './components/Skeleton/Skeleton.js';
export { Slider } from './components/Slider/Slider/Slider.js';
export { RangeSlider } from './components/Slider/RangeSlider/RangeSlider.js';
export { Space } from './components/Space/Space.js';
export { Spoiler } from './components/Spoiler/Spoiler.js';
export { Stack } from './components/Stack/Stack.js';
export { Stepper } from './components/Stepper/Stepper.js';
export { StepperStep } from './components/Stepper/StepperStep/StepperStep.js';
export { StepperCompleted } from './components/Stepper/StepperCompleted/StepperCompleted.js';
export { Switch } from './components/Switch/Switch.js';
export { SwitchGroup } from './components/Switch/SwitchGroup/SwitchGroup.js';
export { Table } from './components/Table/Table.js';
export { TableCaption, TableTbody, TableTd, TableTfoot, TableTh, TableThead, TableTr } from './components/Table/Table.components.js';
export { TableScrollContainer } from './components/Table/TableScrollContainer.js';
export { Tabs } from './components/Tabs/Tabs.js';
export { TabsList } from './components/Tabs/TabsList/TabsList.js';
export { TabsTab } from './components/Tabs/TabsTab/TabsTab.js';
export { TabsPanel } from './components/Tabs/TabsPanel/TabsPanel.js';
export { TagsInput } from './components/TagsInput/TagsInput.js';
export { Text } from './components/Text/Text.js';
export { Textarea } from './components/Textarea/Textarea.js';
export { TextInput } from './components/TextInput/TextInput.js';
export { ThemeIcon } from './components/ThemeIcon/ThemeIcon.js';
export { Timeline } from './components/Timeline/Timeline.js';
export { TimelineItem } from './components/Timeline/TimelineItem/TimelineItem.js';
export { Title } from './components/Title/Title.js';
export { Tooltip } from './components/Tooltip/Tooltip.js';
export { TooltipGroup } from './components/Tooltip/TooltipGroup/TooltipGroup.js';
export { TooltipFloating } from './components/Tooltip/TooltipFloating/TooltipFloating.js';
export { transitions as MANTINE_TRANSITIONS } from './components/Transition/transitions.js';
export { Transition } from './components/Transition/Transition.js';
export { getTransitionProps } from './components/Transition/get-transition-props/get-transition-props.js';
export { TypographyStylesProvider } from './components/TypographyStylesProvider/TypographyStylesProvider.js';
